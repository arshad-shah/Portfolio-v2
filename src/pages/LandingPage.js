import React from 'react';
import { Box, Typography, Button, IconButton, Tooltip } from '@material-ui/core';
import { Link } from 'react-scroll';

import ArrowDownwardRoundedIcon from '@material-ui/icons/ArrowDownwardRounded';
// icons
import GitHubIcon from '@material-ui/icons/GitHub';
import LinkedInIcon from '@material-ui/icons/LinkedIn';
import MailIcon from '@material-ui/icons/Mail';
import useStyles from '../styles/LandingPage.styles';

import PropTypes from 'prop-types';

export default function LandingPage({ isMobile }) {
	const classes = useStyles();
	return (
		<Box className={classes.hero} data-testid="hero">
			<Typography className={classes.landingIntro} variant="h4">
				Hi, My Name Is -
			</Typography>

			<Typography className={classes.landingName} variant="h1">
				Arshad Shah!
			</Typography>

			<Typography className={classes.landingDesc} variant="h4">
				A Third Year <b>Computer Science</b> Student and a <b>Software Engineer</b>.
			</Typography>

			<Box className={classes.Links}>
				<Tooltip title="Github Profile" arrow>
					<IconButton
						data-testid="githubForHero"
						role="button"
						className={classes.socialLinks}
						color="primary"
						aria-label="link to Github account of Arshad shah"
						data-block="linktoGithub"
						href="https://github.com/arshad-shah">
						<GitHubIcon fontSize="large" />
					</IconButton>
				</Tooltip>

				<Tooltip title="LinkedIn Profile" arrow>
					<IconButton
						data-testid="linkedInForHero"
						role="button"
						className={classes.socialLinks}
						color="primary"
						aria-label="link to Linkedin account of Arshad shah"
						data-block="linktoLinkedin"
						href="https://www.linkedin.com/in/arshadshah">
						<LinkedInIcon fontSize="large" />
					</IconButton>
				</Tooltip>

				<Tooltip title="Email Me" arrow>
					<IconButton
						data-testid="mailForHero"
						className={classes.socialLinks}
						color="primary"
						role="button"
						aria-label="link to mail account of Arshad shah"
						data-block="linktoMail"
						href="mailto:arshad@arshadshah.com">
						<MailIcon fontSize="large" />
					</IconButton>
				</Tooltip>
			</Box>
			{!isMobile ? (
				<Tooltip title="Scroll Down" arrow>
					<Link to="aboutme" spy smooth>
						<Button
							data-testid="scrollDown"
							aria-label="scroll down button"
							size="large"
							color="secondary"
							variant="outlined"
							data-block="scrolldown">
							<ArrowDownwardRoundedIcon fontSize="large" />
						</Button>
					</Link>
				</Tooltip>
			) : null}
		</Box>
	);
}

LandingPage.propTypes = {
	isMobile: PropTypes.bool.isRequired,
};
