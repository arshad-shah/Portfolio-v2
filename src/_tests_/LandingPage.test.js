import React from "react";
import { render, screen } from "@testing-library/react";
import LandingPage from "../pages/LandingPage";

describe("LandingPage of the site:", () => {
  it("should render LandingPage with all the information in it", () => {
    const { container } = render(<LandingPage />);
    expect(container).toMatchSnapshot();
  });

  it("should render the Hero text", () => {
    render(<LandingPage />);
    const nameOfArshad = screen.getByText("Arshad Shah!");
    expect(nameOfArshad).toBeInTheDocument();
  });

  it("should render the social links", () => {
    render(<LandingPage />);
    const githubIcon = screen.getByRole("button", {
      name: /link to Github account of Arshad shah/i,
    });
    const linkedInIcon = screen.getByRole("button", {
      name: /link to Linkedin account of Arshad shah/i,
    });
    const mailIcon = screen.getByRole("button", {
      name: /link to mail account of Arshad shah/i,
    });

    expect(githubIcon).toBeInTheDocument();
    expect(linkedInIcon).toBeInTheDocument();
    expect(mailIcon).toBeInTheDocument();
  });

  it("should render the scroll down button", () => {
    render(<LandingPage />);
    const scrollDownButton = screen.getByRole("button", {
      name: /scroll down button/i,
    });
    expect(scrollDownButton).toBeInTheDocument();
  });
});
