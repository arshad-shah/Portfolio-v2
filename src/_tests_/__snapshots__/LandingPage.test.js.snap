// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage of the site: should render LandingPage with all the information in it 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-7 makeStyles-hero-1"
    data-testid="hero"
  >
    <h4
      class="MuiTypography-root makeStyles-landingIntro-2 MuiTypography-h4"
    >
      Hi, My Name Is -
    </h4>
    <h2
      class="MuiTypography-root makeStyles-landingName-3 MuiTypography-h2"
    >
      Arshad Shah!
    </h2>
    <h4
      class="MuiTypography-root makeStyles-landingDesc-4 MuiTypography-h4"
    >
      A Third Year 
      <b>
        Computer Science
      </b>
       Student and a 
      <b>
        Software Engineer
      </b>
      .
    </h4>
    <div
      class="MuiBox-root MuiBox-root-8 makeStyles-Links-6"
    >
      <a
        aria-disabled="false"
        aria-label="link to Github account of Arshad shah"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-socialLinks-5 MuiIconButton-colorPrimary"
        data-block="linktoGithub"
        data-testid="githubForHero"
        href="https://github.com/arshad-shah"
        role="button"
        tabindex="0"
        title="Github Profile"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
      <a
        aria-disabled="false"
        aria-label="link to Linkedin account of Arshad shah"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-socialLinks-5 MuiIconButton-colorPrimary"
        data-block="linktoLinkedin"
        data-testid="linkedInForHero"
        href="https://www.linkedin.com/in/arshadshah"
        role="button"
        tabindex="0"
        title="LinkedIn Profile"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z"
            />
            <rect
              height="12"
              width="4"
              x="2"
              y="9"
            />
            <circle
              cx="4"
              cy="4"
              r="2"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
      <a
        aria-disabled="false"
        aria-label="link to mail account of Arshad shah"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-socialLinks-5 MuiIconButton-colorPrimary"
        data-block="linktoMail"
        data-testid="mailForHero"
        href="mailto:arshad@arshadshah.com"
        role="button"
        tabindex="0"
        title="Email Me"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
            />
            <polyline
              points="22,6 12,13 2,6"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
    <a
      class=""
      title="Scroll Down"
    >
      <button
        aria-label="scroll down button"
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-socialLinks-5 MuiButton-outlinedSecondary MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
        data-block="scrolldown"
        data-testid="scrollDown"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="5"
              y2="19"
            />
            <polyline
              points="19 12 12 19 5 12"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </a>
  </div>
</div>
`;
